webhooks/clusterclass_test.go

```diff
@@ -89,3 +88,1371 @@ func TestClusterClassDefaultNamespaces(t *testing.T) {
+
+	ref := &corev1.ObjectReference{
+		APIVersion: "foo",
+		Kind:       "barTemplate",
+		Name:       "baz",
+		Namespace:  "default",
+	}
+	tests := []struct {
+		name      string
+		in        *clusterv1.ClusterClass
+		old       *clusterv1.ClusterClass
+		expectErr bool
+	}{
+		{
+			name: "creation should fail if feature flag is disabled, no matter the ClusterClass is valid(or not)",
+			in: &clusterv1.ClusterClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+				},
+				Spec: clusterv1.ClusterClassSpec{
+					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
+					ControlPlane: clusterv1.ControlPlaneClass{
+						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: ref},
+					},
+					Workers: clusterv1.WorkersClass{
+						MachineDeployments: []clusterv1.MachineDeploymentClass{
+							{
+								Class: "aa",
+								Template: clusterv1.MachineDeploymentClassTemplate{
+									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: ref},
+									Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
+		{
+			name: "update should fail if feature flag is disabled, no matter the ClusterClass is valid(or not)",
+			old: &clusterv1.ClusterClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+				},
+				Spec: clusterv1.ClusterClassSpec{
+					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
+					ControlPlane: clusterv1.ControlPlaneClass{
+						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: ref},
+					},
+					Workers: clusterv1.WorkersClass{
+						MachineDeployments: []clusterv1.MachineDeploymentClass{
+							{
+								Class: "aa",
+								Template: clusterv1.MachineDeploymentClassTemplate{
+									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: ref},
+									Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
+								},
+							},
+						},
+					},
+				},
+			},
+			in: &clusterv1.ClusterClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+				},
+				Spec: clusterv1.ClusterClassSpec{
+					Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
+					ControlPlane: clusterv1.ControlPlaneClass{
+						LocalObjectTemplate: clusterv1.LocalObjectTemplate{Ref: ref},
+					},
+					Workers: clusterv1.WorkersClass{
+						MachineDeployments: []clusterv1.MachineDeploymentClass{
+							{
+								Class: "aa",
+								Template: clusterv1.MachineDeploymentClassTemplate{
+									Metadata:       clusterv1.ObjectMeta{},
+									Bootstrap:      clusterv1.LocalObjectTemplate{Ref: ref},
+									Infrastructure: clusterv1.LocalObjectTemplate{Ref: ref},
+								},
+							},
+						},
+					},
+				},
+			},
+			expectErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			c := &ClusterClass{}
```

Comment:
```suggestion
			webhook := &ClusterClass{}
```
(similar to cluster_test.go)
Same in l.1450